website/docs/pubsub.md

```diff
@@ -0,0 +1,222 @@
+```
+
+2. Create Dapr pubsub component
+```
+kubectl apply -f <<EOF
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: pubsub
+  namespace: fake-subscriber
+spec:
+  type: pubsub.redis
+  version: v1
+  metadata:
+  - name: redisHost
+    value: redis-master.default.svc.cluster.local:6379
+  - name: redisPassword
+    secretKeyRef: 
+      name: redis
+      key: redis-password
+```
+
+3. Deploy subscriber application
+```
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: sub
+  namespace: fake-subscriber
+  labels:
+    app: sub
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: sub
+  template:
+    metadata:
+      labels:
+        app: sub
+      annotations:
+        dapr.io/enabled: "true"
+        dapr.io/app-id: "subscriber"
+        dapr.io/enable-api-logging: "true"
+        dapr.io/app-port: "6002"
+    spec:
+      containers:
+      - name: go-sub
+        image: fake-subscriber:latest
+        imagePullPolicy: Never
+```
+**Note:** Dockerfile to build image for fake-subscriber is under [gatekeeper/test/fake-subscriber](https://github.com/open-policy-agent/gatekeeper/tree/master/test/pubsub/fake-subscriber). You can find make rule to build and deploy subscriber in [Makefile](../../Makefile) under name `e2e-subscriber-build-load-image` and `e2e-subscriber-deploy`.
+
+#### Configure Gatekeeper with Pubsub enabled
+
+1. Create Dapr pubsub component and redis secret in `gatekeeper-system` (i.e. the namespace where gatekeeper will be installed).
+
+```
+kubectl get secret redis --namespace=default -o yaml | sed 's/namespace: .*/namespace: gatekeeper-system/' | kubectl apply -f -
+kubectl apply -f - <<EOF
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: pubsub
+  namespace: gatekeeper-system
+spec:
+  type: pubsub.redis
+  version: v1
+  metadata:
+  - name: redisHost
+    value: redis-master.default.svc.cluster.local:6379
+  - name: redisPassword
+    secretKeyRef:
+      name: redis
+      key: redis-password
+EOF
+```
+
+2. Install gatekeeper with `--enable-pub-sub` set to `true`, `--audit-connection` set to `audit`, `--audit-channel` set to `audit` on audit pod.
+
+**Note:** Verify that after the audit pod is running there is a dapr sidecar injected and running along side `manager` container.
+
+3. Create connection config to establish a connection.
+
+```
+kubectl apply -f - <<EOF
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: audit
+  namespace: gatekeeper-system
+data:
+  provider: "dapr"
+  config: |
+    {
+      "component": "pubsub"
+    }
+EOF
+```
+**Note:** Name of the connection configMap must match the value of `--audit-connection` for it to be used by audit to publish violation. Hence, right now only one connection config can exists for audit.
+
+4. Create the constraint templates and constraints, and let the audit run it's course.
+
+Finally, check the subscriber logs to see the violations received.
+
+## Pubsub interface and Driver walk through
```

Comment:
I am not sure if these should be in the user-facing docs since these are more dev facing, i am not sure what's the best place to move these though

Reply:
Yeah, I agree. I could not find an appropriate location to add this information.

Reply:
added a page under `Contribute` that describes interface details